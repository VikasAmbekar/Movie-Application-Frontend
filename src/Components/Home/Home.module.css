.headerTop {
  width: 100%;
  height: 4rem;
  background-color: #13363d;
  position: fixed;
  top: 0;
  display: flex;
}

.logo {
  padding: 1.3rem;
  width: 85%;
  text-decoration: none;
  margin-bottom: 15px;
}

.logolink {
  text-decoration: none;
  color: white;
  font-size: 30px;
  left: 0;
  margin-right: 55rem;
  cursor: pointer;
}

.maininfo {
  display: flex;
}

.adminlogin {
  margin-top: 1.3rem;
  width: 15%;
  align-items: right;
  justify-content: right;
}

.adminbtn {
  padding: 0.4rem;
  border: none;
  border-radius: 3px;
  background-color: #c5122d;
  color: white;
  font-size: 14px;
  font-weight: 500;
}

.bottom {
  background-color: gray;
  width: 100%;
  height: 2.5rem;
  position: fixed;
  bottom: 0;
  align-items: center;
  justify-content: center;
  text-align: center;
  padding-bottom: 1rem;
  color: white;
  padding-top: 0.5rem;
}

.carousel {
  width: 60rem;
  height: 21rem;
  /* border: 2px solid black; */
  /* border-radius: 1rem; */
  left: 0;
  margin-top: 4.5rem;
  align-items: center;
  justify-content: center;
  text-align: center;
  margin-left: 1rem;
  border-radius: 0.5px;
}

.carouselImage {
}

.loginForm {
  align-items: center;
  justify-content: center;
  text-align: center;
  /* height: 70vh;
  width: 40%; */

  /* border: 2px solid red; */
  margin-top: 5rem;
  /* margin-left: 0.3rem;
  margin-right: 0.5rem; */
}

.imageSlide {
  border-radius: 6px;
}
