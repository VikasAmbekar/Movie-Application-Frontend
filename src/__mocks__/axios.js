export default{
    get: jest.fn().mockResolvedValue(),
    get: jest.fn().mockResolvedValueOnce()
}