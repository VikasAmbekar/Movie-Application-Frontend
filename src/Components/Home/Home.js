import React, { useEffect, useState } from "react";
import classes from "./Home.module.css";
import { useNavigate } from "react-router-dom";
import { Carousel } from "react-bootstrap";
import UserLogin from "../User/UserLogin";
import axios from "axios";

const Home = () => {
    let navigate = useNavigate();
    const [movies, setMovies] = useState([]);

    async function getMovies() {
        try {
            const response = await axios.get(`http://localhost:8700/movie/`);
            setMovies(response.data);
            console.log(response.data);
            return response.data;
        } catch (error) {
            console.log(error.response);
            throw error;
        }
    }


    useEffect(() => {
        getMovies()
    }, []);

    if (movies === null) {
        return <div></div>;
    }

    return (
        <>
            <header className={classes.headerTop}>
                <div className={classes.logo}>
                    <h5 className={classes.logolink} onClick={() => navigate("/")}>
                        MovieMate
                    </h5>
                </div>
                <div className={classes.adminlogin}>
                    <button
                        className={classes.adminbtn}
                        onClick={() => navigate("/user-signup")}
                    >
                        User SignUp
                    </button>
                </div>
            </header>
            <div className={classes.maininfo}>
                <div className={classes.carousel}>
                    <Carousel fade>
                        {movies.map((movie) => {
                            return (
                                <Carousel.Item className={classes.carouselImage}>
                                    <img
                                        className="d-block"
                                        src={movie.movieImage}
                                        alt="Second slide"
                                        style={{
                                            borderRadius: "10px",
                                            width: "100%",
                                            height: "78vh",
                                        }}
                                    />
                                    <Carousel.Caption>
                                        <h3>{movie.movieTitle}</h3>
                                    </Carousel.Caption>
                                </Carousel.Item>
                            );
                        })}
                    </Carousel>
                </div>
                <div className={classes.loginForm}>
                    <UserLogin />
                </div>
            </div>
            <footer className={classes.bottom}>
                <p>Copyright</p>
            </footer>
        </>
    );
};

export default Home;
