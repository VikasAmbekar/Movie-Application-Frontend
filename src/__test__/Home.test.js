import React from "react";
import {render, waitFor} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Home from "../Components/Home/Home";
import {BrowserRouter as Router} from "react-router-dom";
import axios from "axios";

jest.mock("axios");
describe("Home Component", () => {

    it("matches snapshot", () => {
        const {asFragment} = render(
            <Router>
                <Home/>
            </Router>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders movie carousel with movie images and titles", async () => {
        const mockMovies = [
            { id: 1, movieImage: "movie1.jpg", movieTitle: "Movie 1" },
            { id: 2, movieImage: "movie2.jpg", movieTitle: "Movie 2" },
        ];

        axios.get.mockResolvedValueOnce({ data: mockMovies });

        const { getByAltText } = render(
            <Router>
                <Home />
            </Router>
        );

        await waitFor(() => {
            mockMovies.forEach((movie) => {
                const movieImage = getByAltText(movie.movieTitle);
                expect(movieImage).toBeInTheDocument();
                expect(movieImage).toHaveAttribute("src", movie.movieImage);
            });
        });
    });

    it("handles empty movie array from axios response", async () => {
        axios.get.mockResolvedValueOnce({ data: [] });

        render(
            <Router>
                <Home />
            </Router>
        );

        await waitFor(() => {
            // Assert that there's no carousel items when movies array is empty
            expect(screen.queryByRole("listitem")).not.toBeInTheDocument();
        });
    });

});