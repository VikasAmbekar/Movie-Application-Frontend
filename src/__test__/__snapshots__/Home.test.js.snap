// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="headerTop"
  >
    <div
      class="logo"
    >
      <h5
        class="logolink"
      >
        MovieMate
      </h5>
    </div>
    <div
      class="adminlogin"
    >
      <button
        class="adminbtn"
      >
        User SignUp
      </button>
    </div>
  </header>
  <div
    class="maininfo"
  >
    <div
      class="carousel"
    >
      <div
        class="carousel slide carousel-fade"
      >
        <div
          class="carousel-indicators"
        />
        <div
          class="carousel-inner"
        />
        <a
          class="carousel-control-prev"
          href="#"
          role="button"
          tabindex="0"
        >
          <span
            aria-hidden="true"
            class="carousel-control-prev-icon"
          />
          <span
            class="visually-hidden"
          >
            Previous
          </span>
        </a>
        <a
          class="carousel-control-next"
          href="#"
          role="button"
          tabindex="0"
        >
          <span
            aria-hidden="true"
            class="carousel-control-next-icon"
          />
          <span
            class="visually-hidden"
          >
            Next
          </span>
        </a>
      </div>
    </div>
    <div
      class="loginForm"
    >
      <div>
        <form
          class="loginForm"
        >
          <div>
             
            <br />
            <h2>
               Login Form 
            </h2>
             
            <br />
            <label
              class="loginLabel"
              for="mobile"
            >
              Mobile Number
            </label>
            <br />
            <input
              autocomplete="off"
              class="loginInput"
              id="mobile"
              required=""
              type="tel"
              value=""
            />
          </div>
          <div>
            <label
              for="password"
            >
              Password
            </label>
             
            <br />
            <input
              autocomplete="off"
              class="loginInput"
              id="password"
              required=""
              type="password"
              value=""
            />
          </div>
          <br />
          <button
            class="loginButtom"
            type="submit"
          >
            Login
          </button>
          <br />
          <p
            class="loginSignUpP"
          >
            Don't have account click hereÂ 
            <a
              class="loginSignUp"
            >
              Sign up
            </a>
          </p>
        </form>
      </div>
    </div>
  </div>
  <footer
    class="bottom"
  >
    <p>
      Copyright
    </p>
  </footer>
</DocumentFragment>
`;
